plugin/evm/vm_test.go

```diff
@@ -2566,13 +2580,145 @@ func TestAllowFeeRecipientEnabled(t *testing.T) {
 		t.Fatal(err)
 	}
 
 	balance := blkState.GetBalance(etherBase)
-	assert.Equal(t, 1, balance.Cmp(common.Big0))
+	require.Equal(t, 1, balance.Cmp(common.Big0))
+}
+
+func TestRewardManagerPrecompileSetRewardAddress(t *testing.T) {
+	genesis := &core.Genesis{}
+	require.NoError(t, genesis.UnmarshalJSON([]byte(genesisJSONSubnetEVM)))
+
+	genesis.Config.RewardManagerConfig = precompile.NewRewardManagerConfig(common.Big0, testEthAddrs[0:1], nil, nil)
+	genesis.Config.AllowFeeRecipients = true // enable this in genesis to test if this is recognized by the reward manager
+	genesisJSON, err := genesis.MarshalJSON()
+	require.NoError(t, err)
+
+	etherBase := common.HexToAddress("0x0123456789") // give custom ether base
+	c := Config{}
+	c.SetDefaults()
+	c.FeeRecipient = etherBase.String()
+	configJSON, err := json.Marshal(c)
+	require.NoError(t, err)
+
+	issuer, vm, _, _ := GenesisVM(t, true, string(genesisJSON), string(configJSON), "")
+
+	defer func() {
+		err := vm.Shutdown()
+		require.NoError(t, err)
+	}()
+
+	newTxPoolHeadChan := make(chan core.NewTxPoolReorgEvent, 1)
+	vm.txPool.SubscribeNewReorgEvent(newTxPoolHeadChan)
+
+	testAddr := common.HexToAddress("0x9999991111")
+	data, err := precompile.PackSetRewardAddress(testAddr)
+	require.NoError(t, err)
+
+	tx := types.NewTransaction(uint64(0), precompile.RewardManagerAddress, big.NewInt(1), 21240+precompile.SetRewardAddressGasCost, big.NewInt(testMinGasPrice), data)
+
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.chainConfig.ChainID), testKeys[0])
+	require.NoError(t, err)
+
+	txErrors := vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range txErrors {
+		require.NoError(t, err)
+	}
+
+	blk := issueAndAccept(t, issuer, vm)
+	newHead := <-newTxPoolHeadChan
+	require.Equal(t, newHead.Head.Hash(), common.Hash(blk.ID()))
+	ethBlock := blk.(*chain.BlockWrapper).Block.(*Block).ethBlock
+	require.Equal(t, ethBlock.Coinbase(), etherBase) // reward address is activated at this block so this is fine
+
+	tx1 := types.NewTransaction(uint64(0), testEthAddrs[0], big.NewInt(2), 21000, big.NewInt(testMinGasPrice*3), nil)
+	signedTx1, err := types.SignTx(tx1, types.NewEIP155Signer(vm.chainConfig.ChainID), testKeys[1])
+	require.NoError(t, err)
+
+	txErrors = vm.txPool.AddRemotesSync([]*types.Transaction{signedTx1})
+	for _, err := range txErrors {
+		require.NoError(t, err)
+	}
+
+	blk = issueAndAccept(t, issuer, vm)
+	newHead = <-newTxPoolHeadChan
+	require.Equal(t, newHead.Head.Hash(), common.Hash(blk.ID()))
+	ethBlock = blk.(*chain.BlockWrapper).Block.(*Block).ethBlock
+	require.Equal(t, ethBlock.Coinbase(), testAddr) // reward address was activated at previous block
+	// Verify that etherBase has received fees
+	blkState, err := vm.blockChain.StateAt(ethBlock.Root())
+	require.NoError(t, err)
+
+	balance := blkState.GetBalance(testAddr)
+	require.Equal(t, 1, balance.Cmp(common.Big0))
+}
+
+func TestRewardManagerPrecompileAllowFeeRecipieints(t *testing.T) {
```

Comment:
Can we add the following test cases as well:

- reward manager gets disabled and the rewards go back to the blackhole address
- reward manager gets disable and the rewards go back to genesis configuration of allow fee recipients and miner uses its original specified coinbase address

Reply:
Do you mean tests should cover reward manager precompile itself is disabled or rewards are disabled through the precompile? In the second one I don’t think we should resort back to genesis configuration since it is the intended way to disable whole rewards and send them to blackhole.